### PR TITLE
Always display workspace chats in recents when creating a new request

### PR DESCRIPTION
**Pull request by marcaaron**
### Details

**Doc section here:** 

https://docs.google.com/document/d/17jFcMsAotXvsMhXivoF3u7fyzeyXkKpRGXR88Ud0oWY/edit#bookmark=kix.mwuwbgdkfhxc

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/270684

### Tests
1. Login as a new user (ADMIN) who does not yet have a workspace
2. Create a workspace and invite another new user (EMPLOYEE)
3. Verify the ADMIN can see in the LHN a Workspace Chat (Chat 1) for themselves and a Workspace Chat for the EMPLOYEE
3. Run through the following flows and verify...

Search -> Both Chats appear
LHN -> Both Chats appear
Request money -> Only the user's OWN workspace chat appears (not the managed one)
New chat -> None
New group -> None
Send money -> None
Split bill -> None

1. Login next as the EMPLOYEE
2. Verify they can only see their OWN workspace chat
3. Run through the following flows and verify...

Search -> Own Chat appears
LHN -> Own Chat appears
Request money -> Own Chat appears
New chat -> None
New group -> None
Send money -> None
Split bill -> None

- [ ] Verify that no errors appear in the JS console

### Offline tests
N/A

### QA Steps
Same as Tests

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by marcaaron** [comment]
Still need to test and write up manual QA, but this should be ready I think...

File: src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js [hunk 1]
```diff
@@ -71,6 +67,22 @@ class IOUParticipantsRequest extends Component {
...
+     */
+    getRequestOptions(searchTerm = '') {
+        return OptionsListUtils.getNewChatOptions(
+            this.props.reports,
+            this.props.personalDetails,
+            this.props.betas,
+            searchTerm,
+            [],
+            CONST.EXPENSIFY_EMAILS,
+            this.props.iouType !== CONST.IOU.MONEY_REQUEST_TYPE.SEND,
```
Comment (mountiny) [inline src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js:82]: This would also show the options for Split, right? We dont want that

**Review by mountiny — CHANGES_REQUESTED** [review]

File: src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js (hunk 1 above)
Comment (mountiny) [inline src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js:82]: Updated it in the doc, this should be like this:
```suggestion
            this.props.iouType === CONST.IOU.MONEY_REQUEST_TYPE.REQUEST,
```

@marcaaron Also could we add some explanation to the the `getRequestOptions` request doc why is it true only for request? thank you!

Comment (marcaaron) [reply src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js:82]: This component doesn't get used in that flow.

https://github.com/Expensify/App/blob/e3597c68d03ad48b50b328daf5cf3dbd269b678d/src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsPage.js#L72-L87

Comment (marcaaron) [reply src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js:82]: Sure. This is the same logic though, right?

>could we add some explanation to the the getRequestOptions request doc why is it true only for request?

Maybe better to add that context here - can add something

File: src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js [hunk 2]
```diff
@@ -71,6 +67,26 @@ class IOUParticipantsRequest extends Component {
...
+            this.props.reports,
+            this.props.personalDetails,
+            this.props.betas,
+            searchTerm,
+            [],
+            CONST.EXPENSIFY_EMAILS,
+
+            // If we are using this component in the "Request money" flow then we pass the includeOwnedWorkspaceChats argument so that the current user
+            // sees the option to request money from their admin on their own Workspace Chat. These will always be shown in the "Recents" section of the selector
+            // along with any other recent chats.
```
Comment (Julesssss) [inline src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js:85]: 👍

**Comment by mountiny** [comment]
Yeah i think the comment is great.

We should also keep in mind that contributors have little context about our roadmap and internal terms so the more specific we are the easier it will be for external people to understand whats going on

File: src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js (hunk 1 above)
Comment (mountiny) [reply src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js]: Thanks! Havent realized, still better to use the request. Thanks for updating it

Comment (mountiny) [reply src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js]: Thank you!

**Comment by marcaaron** [comment]
Still testing, but taking this out of WIP.

**Comment by marcaaron** [comment]
Done testing. Ready for review.

File: src/libs/OptionsListUtils.js [hunk 3]
```diff
@@ -491,6 +492,10 @@ function getOptions(reports, personalDetails, {
         const isPolicyExpenseChat = ReportUtils.isPolicyExpenseChat(report);
         const logins = report.participants || [];
 
+        if (isPolicyExpenseChat && report.isOwnPolicyExpenseChat && !includeOwnedWorkspaceChats) {
+            return;
+        }
+
```
Comment (Julesssss) [inline src/libs/OptionsListUtils.js:498]: I know this is pretty self explanitory, but just to make it slightly faster to understand, maybe we could add a comment explaining this logic.

**Comment by mountiny** [comment]
@fedirjh your account should be on the beta now!

File: src/libs/OptionsListUtils.js (hunk 3 above)
Comment (mountiny) [reply src/libs/OptionsListUtils.js:498]: I am on the same boat, as said in another comment I think any of this logic which is more tied with our internal jargon is great to explain in plain english too

File: src/libs/OptionsListUtils.js [hunk 4]
```diff
@@ -723,16 +731,17 @@ function getNewChatOptions(
     searchValue = '',
     selectedOptions = [],
     excludeLogins = [],
+    includeOwnedWorkspaceChats = false,
 ) {
     return getOptions(reports, personalDetails, {
         betas,
         searchInputValue: searchValue.trim(),
         selectedOptions,
-        excludeChatRooms: true,
```
Comment (mountiny) [inline src/libs/OptionsListUtils.js:731]: Why was this removed? @marcaaron

Comment (mountiny) [reply src/libs/OptionsListUtils.js:731]: Unused, I see, nice find!

File: src/libs/OptionsListUtils.js (hunk 3 above)
Comment (marcaaron) [reply src/libs/OptionsListUtils.js:498]: I always welcome a suggestion. I'm usually pretty cautious about adding too much context if the code is self-explanatory. If people need a deeper understanding about how the product works they can always ask us.

Like I started writing this comment...

>Most lists will not display the user's owned Policy Expense Chats (a.k.a. Workspace Chats). These chats are between a workspace member and their workspace admins and used by the workspace member to request money.

But do we need a comment like this next to every mention of "policy expense chat" ?

Comment (marcaaron) [reply src/libs/OptionsListUtils.js:498]: Anyways, let me know what specifically you guys think needs to be here and I will update it.

**Comment by marcaaron** [comment]
Oh hmm lemme fix that archived workspaces thing. I think I'd agree that should not be visible.

File: src/libs/OptionsListUtils.js (hunk 3 above)
Comment (mountiny) [reply src/libs/OptionsListUtils.js:498]: @marcaaron thanks for context. I am happy to leave this one like its now given your comment here explains the property well https://github.com/Expensify/App/pull/16967/files#r1159042123

**Comment by marcaaron** [comment]
Updated

**Comment by mountiny** [comment]
Great find, definitely should not be there

**Review by mountiny — APPROVED** [review]
Looks good to me, @fedirjh waiting for your sign off 🖖

File: src/libs/OptionsListUtils.js (hunk 3 above)
Comment (marcaaron) [reply src/libs/OptionsListUtils.js:498]: The other comment I thought of was...

>If we have a workspace chat and it is the user's own policy expense chat and we don't want to include them then return early

But doesn't strike me as super useful because it just describes what the code does.

Comment (mountiny) [reply src/libs/OptionsListUtils.js:498]: For me personally its just the question of Why is is shown for request but not for send or split which might not be clear for external people. I think what we have there in the param comment is good. Thank you!

Comment (Julesssss) [reply src/libs/OptionsListUtils.js:498]: Yeah, I think we can probably do without the comment after all 👍

**Review by Julesssss — APPROVED** [review]

**Comment by marcaaron** [comment]
@fedirjh repro steps please

**Review by fedirjh — APPROVED** [review]

**Comment by marcaaron** [comment]
Oh I see it... 👀

**Comment by mountiny** [comment]
Did you approve this by accident @fedirjh ?

**Comment by marcaaron** [comment]
Ok cool cool - yeah reproduced it the same way. I don't think we can actually request money from a room yet this is just a step towards the goal.
